site/packages/aa-core/smart-account-client/middleware/index.md

```diff
@@ -57,9 +59,19 @@ This is a no-op middleware, but you can include any custom step during the user
 
 If you are simulating a user operation using [`simulateUserOperation`](/packages/aa-alchemy/smart-account-client/actions/simulateUserOperation.md), `SmartAccountClient` will include `userOperationSimulator` middleware during the pipeline run to simulate the user operation instead of sending it to the network to be mined.
 
-## Use custom middleware using `ClientMiddlewareConfig` in `SmartAccountClientConfig`
+During the `SmartAccountClient` middleware pipeline run, each middleware is applied to compute certain fields of the [`UserOperationStruct`](/resources/types#useroperationstruct) in the following sequential order:
+
+1. `dummyPaymasterDataMiddleware` -- populates a dummy paymaster data to use in estimation (default: `0x` for no paymater)
+2. `feeEstimator` -- sets maxfeePerGas and maxPriorityFeePerGas
+3. `gasEstimator` -- calls eth_estimateUserOperationGas
```

Comment:
I would say the gas estimator "sets verificationGasLimit, callGasLimit, and preVerificationGas". This middleware doesn't _necessarily_ call eth_estimateUserOperationGas, although if you wanted to mention it you could maybe add "likely by calling eth_estimateUserOperationGas" to the above suggestion.